### ISSUE TITLE
DejaVuSerifCondensed.ttf: invalid checksum

### ISSUE THREAD

**Issue by lord-overload** [mpdf/mpdf#1221]
Hi, got strange exeption with MPDF version 8.0.4 and Php version 7.3.17 and 7.4.5:

[Mpdf\MpdfException] E_ERROR
TTF file "/srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Config/../../ttfonts/DejaVuSerifCondensed.ttf": invalid checksum 97c94672 table: FFTM (expected 66091db2) (0)
/srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/TTFontFile.php:335
#0: Mpdf\TTFontFile->checksumTables()
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/TTFontFile.php:316
#1: Mpdf\TTFontFile->readTableDirectory(boolean)
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/TTFontFile.php:3482
#2: Mpdf\TTFontFile->makeSubset(string, array, integer, boolean, boolean)
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Writer/FontWriter.php:287
#3: Mpdf\Writer\FontWriter->writeFonts()
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Writer/ResourceWriter.php:113
#4: Mpdf\Writer\ResourceWriter->writeResources()
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Mpdf.php:9856
#5: Mpdf\Mpdf->_enddoc()
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Mpdf.php:2021
#6: Mpdf\Mpdf->Close()
            /srv/www/<cutted>/libs/MPDF805/vendor/mpdf/mpdf/src/Mpdf.php:9347
#7: Mpdf\Mpdf->Output(string, string)
            /srv/www/<cutted>/report/pdf/cover/title.php:23

Test PHP code produces the same error:
require_once __DIR__ . '/vendor/autoload.php';
$mpdf = new \Mpdf\Mpdf();
$mpdf->WriteHTML('Hello World');
$mpdf->Output();

Mpdf object is created fine, but the exeption occrus when WriteHTML method is called. 

Php version 7.1.30 works fine with the same code.
Platform - Ubuntu 16.04.1.
Any idea?

**Comment by lord-overload** [comment]
Add. info:
Mode => 'en-x' and mode => 'c' wokrs fine with english, but i have to output russian.
Mode => 'ru-x' causes the same error.

**Comment by finwe** [comment]
Your code example is most likely incomplete. This exception is thrown only when `$mpdf->debugfonts` is set to true.

Add a HTML/CSS code example causing this. Try clearing mPDF cache before generating the file.

**Comment by lord-overload** [comment]
Thanx!
You are right, I set `$mpdf->debugfonts` to "true" to find out what is wrong. Browser window was blank and only the message "This pdf document might not be displayed correctly" (highligthed in yellow) was displayed. And there were no errors in php logs.

PHP code:
```
require_once("MPDF805/vendor/autoload.php");
if (!function_exists("init_mpdf")) {
	function init_mpdf() {
		$mpdf = new \Mpdf\Mpdf([
			"mode" => "utf-8",
			"format" => "A4",
		]);
		return $mpdf;
	}
}
$html = "<html>Привет, мир!<br>Hello, world!</html>";
$mpdf = init_mpdf();
$stylesheet = file_get_contents("styles.css");
if ($stylesheet) $mpdf->WriteHTML($stylesheet, 1);
$mpdf->WriteHTML($html);
$mpdf->Output("test.pdf", "I");
```

CSS code:
```
body {
	font-size: 8pt;
	margin: 5pt;
	color: #000000;
}
```

**Comment by finwe** [comment]
I am unable to reproduce this with your code. The PDF is generated correctly without errors.

Are you using canonical version of mPDF (from github/packagist) without changes to fonts/files?

**Comment by lord-overload** [comment]
Yes, the version I am using was installed via composer from github (`composer require mpdf/mpdf`). This code work fine with PHP version 7.1.30. The exeption ocurrs with PHP version 7.3.17.

**Comment by lord-overload** [comment]
Fixed for 7.3.17!
The decision is to set `mbstring.func_overload = 0` globally in php.ini.
I am using Bitrix CMS, which needs `mbstring.func_overload = 2`. For directories containing MPDF code I set `mbstring.func_overload = 0`. That works only with PHP version 7.1.30!
For PHP version 7.3.17  it is necessary to set `mbstring.func_overload = 0` globally (in php.ini), `mbstring.func_overload = 2` for virtual host and `mbstring.func_overload = 0` for directories containing MPDF code (in apache.conf).

**Comment by finwe** [comment]
_username checks out_

That's weird. `mbstring.func_overload` should not interfere with checksums for fonts. 

So: the global failure was caused by the exception thrown when `mbstring.func_overload` is not 0? And trying to debug fonts caused the unrelated checksum mismatch?

I'm going to close this now, but if you ever find out the cause of the checkum mismatch, please share it here.
